tools/model_tools/src/openvino/model_zoo/open_model_zoo.py

```diff
@@ -0,0 +1,193 @@
+            raise ValueError('Path {} to model file does not exist.'.format(model_path))
+
+        if model_path.suffix not in ['.xml', '.onnx']:
+            raise ValueError('Unsupported model format {}. Only .xml or .onnx supported.'.format(model_path.suffix))
+
+        description = 'Pretrained model {}'.format(model_path.name)
+        name = model_path.name
+        task_type = task_type
+        subdirectory = model_path.parent
+        model_path = str(model_path)
+
+        return cls(name, model_path, description, task_type, subdirectory)
+
+    def _load(self, ie, device):
+        self.net = ie.read_network(self.model_path)
+        self.exec_net = ie.load_network(self.net, device)
+
+    def __call__(self, inputs, ie, device='CPU'):
+        if self.exec_net is None:
+            self._load(ie, device)
+
+        input_names = next(iter(self.net.input_info))
+        for input_name, value in inputs.items():
+            if input_name not in input_names:
+                raise ValueError('Unknown input name {}'.format(input_name))
+
+            input_shape = self.net.input_info[input_name].input_data.shape
+            if input_shape != list(value.shape):
+                value = cv2.resize(value, input_shape)
+        res = self.exec_net.infer(inputs=inputs)
+        return res
+
+    @property
+    def accuracy_checker_config(self):
+        if self._accuracy_config is None:
+            config_path = _common.MODEL_ROOT / self.subdirectory / 'accuracy-check.yml'
+            if config_path.exists():
+                with config_path.open('rb') as config_file, \
+                        validation.deserialization_context('Loading config "{}"'.format(config_path)):
+                    self._accuracy_config = yaml.safe_load(config_file)
+
+        return self._accuracy_config
+
+    @property
+    def model_config(self):
+        if self._model_config is None:
+            config_path = _common.MODEL_ROOT / self.subdirectory / 'model.yml'
+            if config_path.exists():
+                with config_path.open('rb') as config_file, \
+                        validation.deserialization_context('Loading config "{}"'.format(config_path)):
+                    self._model_config = yaml.safe_load(config_file)
+
+        return self._model_config
+
+    def inputs(self, ie=None):
+        if self.net is None:
+            try:
+                self.net = ie.read_network(self.model_path)
+            except AttributeError:
+                raise TypeError('ie argument must be of IECore type.')
+
+        input_blob = self.net.input_info
+
+        return input_blob
+
+    def outputs(self, ie=None):
+        if self.net is None:
+            try:
+                self.net = ie.read_network(self.model_path)
+            except AttributeError:
+                raise TypeError('ie argument must be of IECore type.')
+
+        output_blob = self.net.outputs
+
+        return output_blob
+
+    def input_shape(self, name):
```

Comment:
possibly we need more meaningful name for it, preferable_input_shape pr static_input_shape. it is not obvious why we need this method now (real network shape can be different)

Reply:
Renamed to `preferable_input_shape`